package-toolkit/runtime/src/main/kotlin/com/atlan/pkg/serde/cell/DataDomainXformer.kt

```diff
@@ -0,0 +1,59 @@
+/* SPDX-License-Identifier: Apache-2.0
+   Copyright 2023 Atlan Pte. Ltd. */
+package com.atlan.pkg.serde.cell
+
+import com.atlan.model.assets.Asset
+import com.atlan.model.assets.DataDomain
+import com.atlan.model.assets.DataProduct
+import com.atlan.pkg.cache.DataDomainCache
+
+/**
+ * Static object to transform data domain references.
+ */
+object DataDomainXformer {
+
+    const val DATA_DOMAIN_DELIMITER = "@"
+
+    /**
+     * Encodes (serializes) a data domain reference into a string form.
+     *
+     * @param asset to be encoded
+     * @return the string-encoded form for that asset
+     */
+    fun encode(asset: Asset): String {
+        return when (asset) {
+            is DataDomain -> {
+                val dataDomain = DataDomainCache.getByGuid(asset.guid)
+                if (dataDomain is DataDomain) {
+                    DataDomainCache.getIdentity(dataDomain.guid) ?: ""
+                } else {
+                    ""
+                }
+            }
+            else -> AssetRefXformer.encode(asset)
+        }
+    }
+
+    /**
+     * Decodes (deserializes) a string form into a category reference object.
```

Comment:
Copy / paste here refers to `category reference` should be `data domain reference` (maybe other places, too?)

Reply:
fixed.